src/lib/utils.ts

```diff
@@ -47,6 +51,121 @@ export function getMsgIdFromLink(link: string): string {
 	return msgId;
 }
 
+export async function modifyRoleDD(interaction: CommandInteraction, role: Role, isCourse: boolean, dropdownAction: 'ADD' | 'REMOVE'): Promise<boolean> {
+	let rolesMsg: Message;
+	const channel = await interaction.guild.channels.fetch(CHANNELS.ROLE_SELECT) as TextChannel;
+	if (!channel || channel.type !== 'GUILD_TEXT') {
+		const responseEmbed = new MessageEmbed()
+			.setColor('#ff0000')
+			.setTitle('Argument error')
+			.setDescription(`Could not find channel.`);
+		interaction.channel.send({ embeds: [responseEmbed] });
+		return false;
+	}
+	try {
+		rolesMsg = await channel.messages.fetch(
+			isCourse ? ROLE_DROPDOWNS.COURSE_ROLES : ROLE_DROPDOWNS.ASSIGN_ROLES, { cache: true, force: true }
+		);
+	} catch (error) {
+		const responseEmbed = new MessageEmbed()
+			.setColor('#ff0000')
+			.setTitle('Argument error')
+			.setDescription(`Unknown message, make sure your channel and message ID are correct.`);
+		interaction.channel.send({ embeds: [responseEmbed] });
+		return false;
+	}
+	if (rolesMsg.author.id !== BOT.CLIENT_ID) {
+		const responseEmbed = new MessageEmbed()
+			.setColor('#ff0000')
+			.setTitle('Argument error')
+			.setDescription(`You must tag a message that was sent by ${BOT.NAME} (me!).`);
+		interaction.channel.send({ embeds: [responseEmbed] });
+		return false;
+	}
+
+	// the message component row that the dropdown is in
+	let dropdownRow = rolesMsg.components[0] as MessageActionRow;
+	if (!dropdownRow) dropdownRow = new MessageActionRow();
+
+	const option: MessageSelectOptionData = { label: role.name, value: role.id };
+
+	const menu = dropdownRow.components[0] as MessageSelectMenu;
+	switch (dropdownAction) {
+		case 'ADD':
+			return addRole(interaction, rolesMsg, menu, option, dropdownRow, isCourse);
+		case 'REMOVE':
+			return removeRole(interaction, rolesMsg, menu, option, dropdownRow);
+	}
+}
+
+function addRole(interaction: CommandInteraction,
+	rolesMsg: Message,
+	menu: MessageSelectMenu,
+	option: MessageSelectOptionData,
+	dropdownRow: MessageActionRow,
+	isCourse: boolean): boolean {
+	if (menu) {
+		menu.options.forEach(menuOption => {
+			if (menuOption.value === option.value) {
+				const responseEmbed = new MessageEmbed()
+					.setColor('#ff0000')
+					.setTitle('Argument error')
+					.setDescription(`${option.label} is in this message's role select menu already.\nTrying to remove this role from the dropdown? Use /role with the 'Remove' mode 
```

Comment:
```suggestion
					.setDescription(`${option.label} is in this message's role select menu already.
```
sorry I forgot to modify this before handing the code over to you